internal/core/data/coredata_test.go

```diff
@@ -0,0 +1,362 @@
+package data
```

Comment:
I'm still not thrilled with the name of this file. If we can eliminate some of it, then the remainder could simply be put into the first _test.go file in the package.